src/JPEGView/JXLWrapper.cpp

```diff
@@ -0,0 +1,184 @@
+#include "stdafx.h"
+
+#include "JXLWrapper.h"
+#include "jxl/decode.h"
+#include "jxl/decode_cxx.h"
+#include "jxl/resizable_parallel_runner.h"
+#include "jxl/resizable_parallel_runner_cxx.h"
+#include "MaxImageDef.h"
+#include <vector>
+
+JxlDecoderPtr cached_jxl_decoder = NULL;
```

Comment:
I'm not really clear how these variables are scoped.  So this variable appears to be scoped globally more or less, and not tied to a class... but later in JxlReader::DeleteCache() it sets this to NULL?

Does that mean these should be static class members of JxlReader and not file scoped?

Am I reading this code right?

Reply:
Yeah, the caching I did was pretty messy. Would something like this in JXLWrapper.h be better?
```c
struct jxl_cache 
{
	JxlDecoderPtr decoder;
	JxlResizableParallelRunnerPtr runner;
	JxlBasicInfo info;
	uint8_t* data;
	size_t data_size;
	int prev_frame_timestamp;
	int width;
	int height;
};

...

private:
	static jxl_cache cache;
};
```

Reply:
Yeah, that's about what I was thinking, encapsulating that data into the `JxlReader` class seems to be a better approach.  Thanks!